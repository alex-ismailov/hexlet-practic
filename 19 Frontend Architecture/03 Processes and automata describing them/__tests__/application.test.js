import fs from 'fs';
import path from 'path';
import prettier from 'prettier';
import userEvent from '@testing-library/user-event';

// import run from '../src/application.js';
import run from '../src/application teacher solution.js';

const options = {
  parser: 'html',
  htmlWhitespaceSensitivity: 'ignore',
  tabWidth: 4,
};

const fixturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => prettier.format(document.body.innerHTML, options);

beforeEach(() => {
  const initHtml = fs.readFileSync(path.join(fixturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application 1', async () => {
  expect(getTree()).toMatchSnapshot();

  const name = document.querySelector('[data-editable-target="name"]');
  name.click();
  expect(getTree()).toMatchSnapshot();

  const form1 = document.querySelector('form');
  const input1 = form1.elements.name;
  await userEvent.type(input1, 'Hexlet');
  form1.submit();
  expect(getTree()).toMatchSnapshot();

  const email = document.querySelector('[data-editable-target="email"]');
  email.click();
  expect(getTree()).toMatchSnapshot();

  const form2 = document.querySelector('form');
  const input2 = form2.elements.email;
  await userEvent.type(input2, 'support@hexlet.io');
  form2.submit();
  expect(getTree()).toMatchSnapshot();

  name.click();
  const form3 = document.querySelector('form');
  const input3 = form3.elements.name;
  await userEvent.clear(input3);
  form3.submit();
  expect(getTree()).toMatchSnapshot();
});

test('application 2', async () => {
  expect(getTree()).toMatchSnapshot();
});
