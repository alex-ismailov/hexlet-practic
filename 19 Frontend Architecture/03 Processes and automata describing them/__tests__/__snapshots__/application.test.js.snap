// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1 1`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\"><i>name</i></div>
    <div data-editable-target=\\"email\\"><i>email</i></div>
</div>
"
`;

exports[`application 1 2`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">
        <form>
            <input type=\\"text\\" name=\\"name\\" value=\\"\\" />
            <input type=\\"submit\\" value=\\"Save\\" />
        </form>
    </div>
    <div data-editable-target=\\"email\\"><i>email</i></div>
</div>
"
`;

exports[`application 1 3`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">Hexlet</div>
    <div data-editable-target=\\"email\\"><i>email</i></div>
</div>
"
`;

exports[`application 1 4`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">Hexlet</div>
    <div data-editable-target=\\"email\\">
        <form>
            <input type=\\"text\\" name=\\"email\\" value=\\"\\" />
            <input type=\\"submit\\" value=\\"Save\\" />
        </form>
    </div>
</div>
"
`;

exports[`application 1 5`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\">Hexlet</div>
    <div data-editable-target=\\"email\\">support@hexlet.io</div>
</div>
"
`;

exports[`application 1 6`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\"><i>name</i></div>
    <div data-editable-target=\\"email\\">support@hexlet.io</div>
</div>
"
`;

exports[`application 2 1`] = `
"<div class=\\"container m-3\\">
    <div data-editable-target=\\"name\\"><i>name</i></div>
    <div data-editable-target=\\"email\\"><i>email</i></div>
</div>
"
`;
